internal/host/transition.go

```diff
@@ -448,13 +448,31 @@ func (th *transitionHandler) HasInstallationTimedOut(sw stateswitch.StateSwitch,
 	if !ok {
 		return false, errors.New("HasInstallationTimedOut incompatible type of StateSwitch")
 	}
+	params, ok := args.(*TransitionArgsRefreshHost)
+	if !ok {
+		return false, errors.New("HasInstallationTimedOut invalid argument")
+	}
+	if isPaused, err := IsInstallationPendingUserAction(sHost.host.ClusterID, params.db); err != nil {
+		return false, err
+	} else if isPaused {
+		return false, nil
+	}
 	return time.Since(time.Time(sHost.host.StatusUpdatedAt)) > InstallationTimeout, nil
 }
 
 func (th *transitionHandler) HasInstallationInProgressTimedOut(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) (bool, error) {
 	sHost, ok := sw.(*stateHost)
 	if !ok {
-		return false, errors.New("HasInstallationTimedOut incompatible type of StateSwitch")
+		return false, errors.New("HasInstallationInProgressTimedOut incompatible type of StateSwitch")
+	}
+	params, ok := args.(*TransitionArgsRefreshHost)
+	if !ok {
+		return false, errors.New("HasInstallationInProgressTimedOut invalid argument")
+	}
+	if isPaused, err := IsInstallationPendingUserAction(sHost.host.ClusterID, params.db); err != nil {
```

Comment:
IMO this validation should be relevant for specific stages of the installation, for example writing to disk can still timeout and it's not related to wrong boot order or any user action. So maybe need to check the stage as well. 
waiting for control plane can be affected by this state.
@avishayt @eranco74 what do you think?

Reply:
I don't understand.  How can the host be in pending-user-action while writing to disk?  It should only happen in the reboot stage.

Reply:
This is exactly what i'm saying, this validation should be done on specific stages of the installation.

Reply:
As an optimization?

Reply:
No, it will hide bugs, or at least delay detection.

Reply:
I don't see how but I may be missing some nuance here.  I assumed that if we're in a different stage, then we're not in pending-user-action state, and the check will return false.

Reply:
waiting for user action can happen if one of the hosts is having a wrong boot order. So if it's a master then other hosts will not be able to connect to control plane.
This validation checking only cluster state, and not the current stage of the host. 
So the validation should be that cluster is in `IsInstallationPendingUserAction` and the current installation stage is one of 
1. Waiting for control plane
2. Start waiting for control plane
3. Rebooting